tests/plugins/dialog/plugin.js

```diff
@@ -401,6 +432,22 @@
 
 			editor.execCommand( 'testDialog6' );
 			wait();
+		},
+
+		// (#2277)
+		'test default values': function() {
+			var tc = this;
+			this.editor.openDialog( 'testDialog1', function( dialog ) {
+				tc.resume( function() {
+					wait( function() {
+						assert.areEqual( 'text1', dialog.getContentElement( 'info', 'text1' ).getValue(), 'text1 field has invalid value' );
+						assert.areEqual( 'text2', dialog.getContentElement( 'info', 'text2' ).getValue(), 'text2 field has invalid value' );
+
+						dialog.getButton( 'cancel' ).click();
+					}, 100 );
```

Comment:
Do we really need to do a nasty timeout here? 😢 

There's `editorBot.dialog` which would simplify that. It actually uses proper(ish) event to resume, though still it uses 0 ms timeout - but that's much better than that.